indexer/indices/letter.py

```diff
@@ -0,0 +1,139 @@
+from elasticsearch import helpers
+
+from indexer.firstletter import FirstLetter
+from indexer.indices.base import BaseIndex
+from indexer.indices.errors import IndexationError
+
+
+class LetterIndex(BaseIndex):
+    """
+    Creates a Letter index allowing to check verbs based off their first letter.
+
+    Args:
+        es_url (str): Elasticsearch instance url. Defaults to 'http://localhost:9200'.
+    """
+
+    def __init__(self, es_url: str = "http://localhost:9200") -> None:
+        """
+        Initializes a LetterIndex instance for the letter feature
+        based off Elasticsearch.
+
+        Args:
+            es_url (str): Elasticsearch instance url. Defaults to 'http://localhost:9200'.
+        """
+        super().__init__(es_url)
+        self.index_name = "letter-index"
+        self.letter = FirstLetter()
+        self.entries = 0
+        self.bulk_ops = []
+
+    def _build_mapping(self) -> dict:
+        """
+        Builds the necessary mappings for creating an index for the letter feature.
+        Creates the following fields:
+            - verb_form
+            - infinitive
+            - index_letter
+        Also attaches a custom tokenizer based on a regex pattern and a custom analyzer for
+        checking every term in lowercase.
+
+        Returns:
+            dict: A dictionary containing the mappings to create an index.
+        """
+        return {
```

Comment:
My understanding is that settings is the same across the 3 index classes.
Consider if we can just define this part once in the base class and reuse in the other.
Prevents duplicating this configuration 3 times.

Reply:
That is correct, they are very similar but not exactly the same. Let's leave it then as it is. Thanks